modules/fluid_properties/src/userobjects/Water97FluidProperties.C

```diff
@@ -0,0 +1,1411 @@
+   switch (region)
+   {
+     case 1:
+     {
+       pi = pressure / _p_star[0];
+       tau = _T_star[0] / temperature;
+       Real dgdp = dgamma1_dpi(pi, tau);
+       Real d2gdpt = d2gamma1_dpitau(pi, tau);
+       dinternal_energy_dp = _Rw * temperature * (tau * d2gdpt - dgdp - pi *
+         d2gamma1_dpi2(pi, tau)) / _p_star[0];
+       dinternal_energy_dT = _Rw * (tau * tau * d2gamma1_dtau2(pi, tau) - pi *
+         dgdp - pi * tau * d2gdpt);
+       break;
+     }
+
+     case 2:
+     {
+       pi = pressure / _p_star[1];
+       tau = _T_star[1] / temperature;
+       Real dgdp = dgamma2_dpi(pi, tau);
+       Real d2gdpt = d2gamma2_dpitau(pi, tau);
+       dinternal_energy_dp = _Rw * temperature * (tau * d2gdpt - dgdp - pi *
+         d2gamma2_dpi2(pi, tau)) / _p_star[1];
+       dinternal_energy_dT = _Rw * (tau * tau * d2gamma2_dtau2(pi, tau) - pi *
+         dgdp - pi * tau * d2gdpt);
+       break;
+     }
+
+     case 3:
+     {
+       // Calculate density first, then use that in Helmholtz free energy
+       Real density3 = densityRegion3(pressure, temperature);
+       Real delta = density3 / _rho_critical;
+       tau = _T_star[2] / temperature;
+       dinternal_energy_dp = _Rw * _T_star[2] * d2phi3_ddeltatau(delta, tau) /
+         _rho_critical / (2.0 * _Rw * temperature * delta * dphi3_ddelta(delta, tau) +
+         _Rw * temperature * delta * delta * d2phi3_ddelta2(delta, tau));
+       dinternal_energy_dT = - _Rw * tau * tau * ( d2phi3_dtau2(delta, tau));
+       break;
+     }
+
+     case 5:
+     {
+       pi = pressure / _p_star[4];
+       tau = _T_star[4] / temperature;
+       Real dgdp = dgamma5_dpi(pi, tau);
+       Real d2gdpt = d2gamma5_dpitau(pi, tau);
+       dinternal_energy_dp = _Rw * temperature * (tau * d2gdpt - dgdp - pi *
+         d2gamma5_dpi2(pi, tau)) / _p_star[4];
+       dinternal_energy_dT = _Rw * (tau * tau * d2gamma5_dtau2(pi, tau) - pi *
+         dgdp - pi * tau * d2gdpt);
+       break;
+     }
+    }
+
+  e = this->e(pressure, temperature);
+  // Divide by 1000 as output in kJ/kg
+  de_dp = dinternal_energy_dp / 1000.0;
+  de_dT = dinternal_energy_dT / 1000.0;
+}
+
+void
+Water97FluidProperties::rho_e_dpT(Real pressure, Real temperature, Real & rho, Real & drho_dp, Real & drho_dT, Real & e, Real & de_dp, Real & de_dT) const
+{
+  rho_dpT(pressure, temperature, rho, drho_dp, drho_dT);
+  e_dpT(pressure, temperature, e, de_dp, de_dT);
+}
+
+Real
+Water97FluidProperties::c(Real pressure, Real temperature) const
+{
+  Real speed2, pi, tau, delta;
+
+   // Determine which region the point is in
+   unsigned int region = inRegion(pressure, temperature);
+   switch (region)
+   {
+     case 1:
+       pi = pressure / _p_star[0];
+       tau = _T_star[0] / temperature;
+       speed2 = _Rw * temperature * std::pow(dgamma1_dpi(pi, tau), 2.0) /
+       (std::pow(dgamma1_dpi(pi, tau) - tau * d2gamma1_dpitau(pi, tau), 2.0) /
+       (tau * tau * d2gamma1_dtau2(pi, tau)) - d2gamma1_dpi2(pi, tau));
```

Comment:
Indent these 2 lines by 2 more spaces, so it is clear they are the continuation of the long expression from 3 lines above.